scripts/buyAuctionLands.js

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env babel-node
+
+import fs from 'fs'
+import { eth, Contract } from 'decentraland-eth'
+import { Log, cli, utils } from 'decentraland-commons'
+
+import { connectEth } from '../src/ethereum'
+import { asyncBatch } from '../src/lib'
+import { Bounds } from '../shared/map'
+import { buildCoordinate, splitCoodinatePairs } from '../shared/coordinates'
+import { loadEnv, asSafeAction } from './utils'
+
+const log = new Log('buyAuctionLands')
+const DEFAULT_OPTIONS = {}
+const requiredOptionNames = ['parcels', 'manaTokenAddress']
+
+const buyAuctionLands = {
+  addCommands(program) {
+    program
+      .command('buy')
+      .option('--parcels [parcels]', 'List of parcels to deploy. Required')
+      .option(
+        '--manaTokenAddress [manaTokenAddress]',
+        'MANAToken contract address. Required'
+      )
+      .option('--password [password]', 'Password for the account')
+      .option(
+        '--retryFailedTxs',
+        'If this flag is present, the script will try to retry failed transactions'
+      )
+      .action(
+        asSafeAction(async userOptions => {
+          const options = Object.assign(DEFAULT_OPTIONS, userOptions)
+          checkRequiredOptions(options, requiredOptionNames)
+
+          log.info('Connecting to the blockchain')
+          await connectEth()
+
+          const allParcels = options.parcels
+            ? readJSON(expandPath(options.parcels))
+            : []
+
+          log.info('Checking for invalid parcels')
+          const parcels = await getValidParcels(allParcels)
+
+          if (parcels.length === 0) {
+            log.info('No parcels to bid')
+            return
+          }
+
+          if (options.password) {
+            log.info(`Unlocking account ${eth.getAddress()}`)
+            await eth.wallet.unlockAccount(options.password)
+          }
+
+          await bidOnParcels(
+            parcels,
+            options.manaTokenAddress,
+            options.retryFailedTxs
+          )
+        })
+      )
+  }
+}
+
+async function getValidParcels(allParcels) {
+  const parcels = []
+  const landRegistryContract = eth.getContract('LANDRegistry')
+
+  for (const parcel of allParcels) {
+    const { x, y } = parcel
+    const parcelId = buildCoordinate(x, y)
+
+    if (!Bounds.inBounds(x, y)) {
+      log.info(`Parcel ${parcelId} is out of bounds`)
+      continue
+    }
+
+    const owner = await landRegistryContract.ownerOfLand(x, y)
+    if (!Contract.isEmptyAddress(owner)) {
+      log.info(
+        `Parcel ${parcelId} already has an owner on the blockchain: ${owner}`
+      )
+      continue
+    }
+    parcels.push(parcel)
+  }
+
+  return parcels
+}
+
+async function bidOnParcels(parcels, tokenAddress, shouldRetry = false) {
+  const landAuctionContract = eth.getContract('LANDAuction')
+  const account = eth.getAddress()
```

Comment:
Is it possible to use a different account?

Reply:
I could add a `--account` and use it to call `web3.unlockAccount` directly and on the txs `from`. I thought it wasn't going to be necessary